code/WorkInProgress/recycling/conveyor.dm

```diff
@@ -764,6 +970,56 @@ ADMIN_INTERACT_PROCS(/obj/machinery/conveyor_switch, proc/trigger)
 				C.setdir()
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_SIGNAL,"switchTriggered")
 
+/obj/machinery/conveyor_switch/built/
+	desc = "A conveyor control switch. This one looks like it was built recently."
+
+
+	New()
+		. = ..()
+		UnsubscribeProcess()
+		START_TRACKING
+		UpdateIcon()
+		AddComponent(/datum/component/mechanics_holder)
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"trigger", PROC_REF(trigger))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Refresh ID", PROC_REF(get_new_id))
+		src.get_new_id()
+
+	proc/get_new_id()
+		src.id = TIME // Create a new ID on the fly. It doesn't need to be readable, it just needs to be unique.
+		src.conveyors = list()
+
+	MouseDrop_T(dropped, mob/user)
+		if (ispulsingtool(dropped))
```

Comment:
This checks if the item being dropped is a multitool, perhaps you wanted something like this instead?
```suggestion
		if (user.find_tool_in_hand(TOOL_PULSING))
```